bug: FilterTest#testClassCastExceptionOnForEach seems to fail on CI

See: https://ci.inria.fr/sos/job/Spoon%20Install%20Commit%20Hook/224/console

> Tests run: 40, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 4.194 sec <<< FAILURE! - in spoon.test.filters.FilterTest
testClassCastExceptionOnForEach(spoon.test.filters.FilterTest)  Time elapsed: 0.109 sec  <<< ERROR!
spoon.SpoonException: Execution of query callback failed
	at spoon.test.filters.FilterTest.testClassCastExceptionOnForEach(FilterTest.java:884)
Caused by: java.lang.ClassCastException: spoon.reflect.CtModelImpl$CtRootPackage cannot be cast to spoon.reflect.declaration.CtType
	at spoon.test.filters.FilterTest.testClassCastExceptionOnForEach(FilterTest.java:884)

--- a/src/test/java/spoon/test/filters/FilterTest.java
+++ b/src/test/java/spoon/test/filters/FilterTest.java
@@ -868,6 +868,7 @@ public class FilterTest {
 	@Test
 	public void testClassCastExceptionOnForEach() throws Exception {
 		// contract: bound query, without any mapping
+		// This test could fail with a version of JDK <= 8.0.40. 
 
 		final Launcher launcher = new Launcher();
 		launcher.setArgs(new String[] {"--output-type", "nooutput","--level","info" });

